src/utils/common_utils.cpp

```diff
@@ -55,5 +56,20 @@ bool make_path(const std::string& path) {
     }
 }
 
+std::string generate_random_string(const int len) {
+    std::string s;
+    static const char alphanum[] =
+        "0123456789"
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+        "abcdefghijklmnopqrstuvwxyz";
+    std::random_device dev;
+    std::mt19937 rng(dev());
+    std::uniform_int_distribution<std::mt19937::result_type> dist(0, (sizeof(alphanum) - 1));
+    for (int i = 0; i < len; ++i) {
+        s.push_back(alphanum[dist(rng)]);
```

Comment:
```suggestion
        s[i] = alphanum[dist(rng)];
```
just to avoid those `push_back`s

Reply:
Yeah, good catch :+1: